libs/librrgraph/src/base/rr_spatial_lookup.cpp

```diff
@@ -55,27 +55,27 @@ RRNodeId RRSpatialLookup::find_node(int layer,
         return RRNodeId::INVALID();
     }
 
-    if (size_t(layer) >= rr_node_indices_[type].dim_size(0)) {
+    if (size_t(layer) >= rr_node_indices_[(size_t)type].dim_size(0)) {
         return RRNodeId::INVALID();
     }
 
-    if (node_x >= rr_node_indices_[type].dim_size(1)) {
+    if (node_x >= rr_node_indices_[(size_t)type].dim_size(1)) {
         return RRNodeId::INVALID();
     }
 
-    if(node_y >= rr_node_indices_[type].dim_size(2)){
+    if(node_y >= rr_node_indices_[(size_t)type].dim_size(2)){
         return RRNodeId::INVALID();
     }
 
-    if (node_side >= rr_node_indices_[type].dim_size(3)) {
+    if (node_side >= rr_node_indices_[(size_t)type].dim_size(3)) {
         return RRNodeId::INVALID();
     }
 
-    if (size_t(ptc) >= rr_node_indices_[type][layer][node_x][node_y][node_side].size()) {
+    if (size_t(ptc) >= rr_node_indices_[(size_t)type][layer][node_x][node_y][node_side].size()) {
         return RRNodeId::INVALID();
     }
 
-    return rr_node_indices_[type][layer][node_x][node_y][node_side][ptc];
+    return rr_node_indices_[(size_t)type][layer][node_x][node_y][node_side][ptc];
```

Comment:
Same comment as above. Can we use vtr vector/matrix to be able use node type directly to index it?